### PR TITLE
[SofaNG] Setup Diffusion

### PR DESCRIPTION
**Pull request by fredroy**
#1527 

Based on master

Deprecates partially SofaSimpleFem

Isolate the diffusion into its own module, especially to highlight that SOFA can do more than solid mechanics.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Component/Diffusion/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+cmake_minimum_required(VERSION 3.12)
+project(Sofa.Component.Diffusion LANGUAGES CXX)
+
+set(SOFACOMPONENTDIFFUSION_SOURCE_DIR "src/sofa/component/diffusion")
+
+set(HEADER_FILES
+	${SOFACOMPONENTDIFFUSION_SOURCE_DIR}/config.h.in
```
Comment (alxbilger) [inline Component/Diffusion/CMakeLists.txt]: indentation

Comment (fredroy) [reply Component/Diffusion/CMakeLists.txt]: What indentation ❓

Comment (alxbilger) [reply Component/Diffusion/CMakeLists.txt]: The way I see it in github, line 7 is not indented like the following lines

Comment (fredroy) [reply Component/Diffusion/CMakeLists.txt]: weird i dont see it in the diff here, but I view the file directly I see it.
thanks !

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp [hunk 2]
```diff
@@ -818,6 +818,9 @@ const std::map< std::string, CreatableMoved, std::less<> > movedComponents =
     { "RecordedCamera", CreatableMoved("v22.06", "SofaGeneralVisual", "Sofa.Component.Visual") },
     { "Visual3DText", CreatableMoved("v22.06", "SofaGeneralVisual", "Sofa.Component.Visual") },
     { "VisualTransform", CreatableMoved("v22.06", "SofaGeneralVisual", "Sofa.Component.Visual") },
+
+    // SofaSimpleFem was deprecated in #XXXX and ....
```
Comment (alxbilger) [inline SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp]: missing PR number

**Review by alxbilger — APPROVED** [review]

**Comment by fredroy** [comment]
add pr number in component change

**Comment by alxbilger** [comment]
@fredroy Could you resolve the conflicts before merging?

**Pull request merged**